### PR TITLE
BD-2642 Migration guide for Google cloud messaging api deprecation

### PR DESCRIPTION
**Pull request by internetisaiah**
# Pull Request/Issue Resolution

#### Description of Change:

Migration guide for Google cloud messaging api depreciation.

- **Jira Ticket:** [BD-2642](https://jira.braze.com/browse/BD-2642)

#### Is this change associated with a Braze feature/product release?
- [x] Yes (**Insert Feature Release Date Here**)
- [ ] No

---

<details>
<summary>✔️ Pull Request Checklist</summary>
<br>

- [ ] Check that you haven't removed any images (replacing an image with an updated one of the same name is fine), as this breaks the French site
- [ ] Check that all links work.
- [ ] Ensure you have completed [our Contributors License Agreement](https://www.braze.com/docs/cla/).
- [ ] Tag @josh-mccrowell-braze and @bre-fitzgerald as a reviewer when your work is **done and ready to be reviewed for merge**. Are you an internal product manager? Reference the internal reviewing chart to tag the appropriate reviewer.
- [ ] If the documentation involves a 1) paid SKU, 2) a third party, 3) SMS, 4) AI, or 5) privacy, ensure that Maria Maldonado on the Legal team has signed off.
- [ ] Tag others as reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `assets` > `js` > `broken_redirect_list.js`

</details>

<details>
<summary>⭐ Helpful Wiki Shortcuts</summary>
<br>

- [Writing Style Guide](https://docs.google.com/document/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub)
- [Image Style Guide](https://docs.google.com/document/d/e/2PACX-1vRJSkwcjmjrTfLDagZccLpOMMyh5NN5SXRZSjz12cRAHbX4OrUmhvCmYpf_p5YB-9r4_jSOQLkicQIH/pub)
- [Styling Test Page](https://www.braze.com/docs/home/styling_test_page/)

</details>

<details>
<summary>❗ ATTN: For PR Reviewers</summary>
<br>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Read our [Previewing Documentation page](https://github.com/braze-inc/braze-docs/wiki/Previewing-and-Testing-Documentation) to see how to check the deployment.
  - [ ] Preview all changes in the linked Vercel environment by clicking the preview link in the vercel-bot comment in your PR.
</details>

<details>
<summary>❗ ATTN: Internal Reviewing Chart </summary>
<br>
<b>Work at Braze and not sure who to tag for review?</b> <br>Before tagging @josh-mccrowell-braze or @bre-fitzgerald for a general review, reference the following chart to see if a specific product vertical/reviewer applies to your pull request.
<br><br>
<table>
<tr>
    <td><b>Reviewer</b></td>
    <td><b>Product Vertical</b></td>
  </tr>
  <tr>
    <td>@josh-mccrowell-braze</td>
    <td>Monolith Deployments<br>Quality Infrastructure<br>Platform Infrastructure<br>Datalake<br>SDKs<br>Currents</td>
  </tr>
  <tr>
    <td>@bre-fitzgerald</td>
    <td>Intelligence<br>In-App Messages<br>Channels<br>FIX</td>
  </tr>
  <tr>
    <td>@lydia-xie</td>
    <td>Ingestion<br>Core Objects<br>Core Messaging<br>Messaging Experience<br>Message Components<br>Email (Composition and Infrastructure) (tag @rachel-feinberg for Liquid use cases)</td>
  </tr>
  <tr>
    <td>@rachel-feinberg</td>
    <td>Customer Lifecycle, Identity and Permissions<br>SMS<br>User Targeting<br>Reporting</td>
  </tr>
</table>
</details>

### CODE REVIEW

File: _docs/_developer_guide/platform_integration_guides/android/push_notifications/android/migrating_to_firebase_cloud_messaging.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  "client_id": "1230498710982375089",
+  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+  "token_uri": "https://oauth2.googleapis.com/token",
+  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/firebase-cloud-messaging.iam.gserviceaccount.com",
+  "universe_domain": "googleapis.com"
+}
+```
+
+Copy the value assigned to `private_key` (excluding the quotes). Your clipboard contents should look similar to the following:
```
Comment (josh-mccrowell-braze) [inline _docs/_developer_guide/platform_integration_guides/android/push_notifications/android/migrating_to_firebase_cloud_messaging.md:88]: This stuff is great. Big believer of this granularity.

**Review by josh-mccrowell-braze — APPROVED** [review]
Left a few comments, but looks good

File: _docs/_developer_guide/platform_integration_guides/android/push_notifications/android/migrating_to_firebase_cloud_messaging.md (hunk 1 above)
Comment (internetisaiah) [reply _docs/_developer_guide/platform_integration_guides/android/push_notifications/android/migrating_to_firebase_cloud_messaging.md:88]: Thanks. As a note to give credit where credit is due, my implementation is heavily inspired by Google's Developer Documentation Style Guide concerning [command-line syntax](https://developers.google.com/style/code-syntax).

File: _docs/_developer_guide/platform_integration_guides/android/push_notifications/android/integration/standard_integration.md [hunk 2]
```diff
@@ -15,48 +15,52 @@ search_rank: 3
...
 
 ![A project's service account home page with "Create Service Account" highlighted.]({% image_buster /assets/img/android/push_integration/create_a_service_account/select-create-service-account.png %})
 
 Enter a service account name, ID, and description, then select **Create and continue**.
 
 ![The form for "Service account details."]({% image_buster /assets/img/android/push_integration/create_a_service_account/enter-service-account-details.png %})
 
-In the **Role** field, find and select **Firebase Cloud Messaging API Admin** from the list of roles. If you need to grant specific users access to your FCM service account, select **Continue**. Otherwise, select **Done**.
+In the **Role** field, find and select **Firebase Cloud Messaging API Admin** from the list of roles. To use a [custom role](https://cloud.google.com/iam/docs/creating-custom-roles) instead, give your role the `cloudmessaging.messages.create` permission, then choose your role from the list. When you're finished, select **Done**.
```
Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/android/push_notifications/android/integration/standard_integration.md:57]:
```suggestion
In the **Role** field, find and select **Firebase Cloud Messaging API Admin** from the list of roles. To use a more restrictive [custom role](https://cloud.google.com/iam/docs/creating-custom-roles) instead, give your role the `cloudmessaging.messages.create` permission, then choose your role from the list. When you're finished, select **Done**.
```

File: _docs/_developer_guide/platform_integration_guides/android/push_notifications/android/integration/standard_integration.md [hunk 3]
```diff
@@ -68,6 +72,10 @@ Choose **JSON**, then select **Create**. Be sure to remember where you downloade
 
 ![The form for creating a private key with "JSON" selected.]({% image_buster /assets/img/android/push_integration/generate_json_credentials/select-create.png %}){: style="max-width:65%;"}
 
+{% alert warning %}
+Private keys could pose a security risk if compromised. Store your JSON credentials in a secure location before continuing.
```
Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/android/push_notifications/android/integration/standard_integration.md:76]:
```suggestion
Private keys could pose a security risk if compromised. Store your JSON credentials in a secure location before continuing, and be sure to delete it from your computer after you've uploaded it to Braze.
```

File: _docs/_developer_guide/platform_integration_guides/android/push_notifications/android/integration/standard_integration.md [hunk 4]
```diff
@@ -15,48 +15,52 @@ search_rank: 3
 
 With push notifications, you can re-engage your app users by sending time-sensitive and relevant content directly to their device screen&#8212;even if their app is closed. When you're finished integrating push for your app, be sure to check out our [push best practices]({{site.baseurl}}/user_guide/message_building_by_channel/push/best_practices/).
 
+{% alert important %}
+If your Android push integration is already set up, and you're looking to migrate from Google's deprecated Cloud Messaging API, see [Migrating to Firebase Cloud Messaging]({{site.baseurl}}/developer_guide/platform_integration_guides/android/push_notifications/android/migrating_google_messaging_api).
```
Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/android/push_notifications/android/integration/standard_integration.md:19]: this link is broken
```suggestion
If your Android push integration is already set up, and you're looking to migrate from Google's deprecated Cloud Messaging API, see [Migrating to Firebase Cloud Messaging]({{site.baseurl}}/developer_guide/platform_integration_guides/android/push_notifications/android/migrating_google_messaging).
```

**Review by davidbielik — APPROVED** [review]
after broken link is fixed, LGTM - thanks!

**Pull request merged**